src/components/MultiHopTrade/hooks/useSupportedAssets.tsx

```diff
@@ -22,9 +26,13 @@ export const useSupportedAssets = () => {
   useEffect(() => {
     ;(async () => {
       const assetIds = await getSupportedSellAssetIds(enabledSwappers)
-      setSupportedSellAssetIds(assetIds)
+      const filteredAssetIds = [...assetIds].filter(assetId => {
```

Comment:
rare feedback - very allocatey and unnecessary spread here?

Reply:
Filter returns a new copy of the array so spread might not be required

Reply:
due to supported sell assets being passed around as a `Set`. most efficient would be `.forEach`, and `filteredAssetIds.add()`

Reply:
She's a set, but yes a `forEach` would reduce the space complexity.

Reply:
Optimised in https://github.com/shapeshift/web/pull/5088/commits/84d5b11d9fbad7a3d8b135fc6ff8c74869c894f4